tests/test_resource.py

```
@@ -200,3 +203,146 @@ def test_proxy_precedence(self, fake_client):
+
+    def test_save_kubeconfig_insufficient_data(self, tmp_path):
+        kubeconfig_path = str(tmp_path / "kubeconfig")
+
+        save_kubeconfig(path=kubeconfig_path)
+
+        assert not os.path.exists(kubeconfig_path)
+
+    def test_save_kubeconfig_file_permissions(self, tmp_path):
+        kubeconfig_path = str(tmp_path / "kubeconfig")
+
+        save_kubeconfig(path=kubeconfig_path, host="https://api.example.com:6443", token="test-token")
```

Ruff (0.15.6)

[error] Possible hardcoded password assigned to argument: "token"

(S106)